usl_pipeline/usl_lib/usl_lib/shared/geo_data.py

```diff
@@ -101,3 +101,110 @@ def to_tuple(self) -> Tuple[float, float, float, float]:
     def from_tuple(bbox: Tuple[float, float, float, float]) -> "BoundingBox":
         """Creates an instance from a Tuple[min_x, min_y, max_x, max_y]."""
         return BoundingBox(min_x=bbox[0], min_y=bbox[1], max_x=bbox[2], max_y=bbox[3])
+
+
+@dataclass
```

Comment:
always recommend setting `slots=True` for dataclasses
would additionally suggest `frozen=True` since these look like they should be immutable
https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass

Reply:
Done